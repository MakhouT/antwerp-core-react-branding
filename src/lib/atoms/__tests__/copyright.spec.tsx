import * as React from "react";

import { shallow } from "enzyme";

import * as A from "../..";
import { renderAndMatchSnapshot } from "../../../__tests__/__fixtures__/testUtils";

describe("Copyright", () => {
    it("Renders default", () => {
        // Arrange
        const element = <A.Copyright name="Rombit" />;

        // Act and assert
        renderAndMatchSnapshot(element);
    });

    it("Renders with correct text", () => {
        // Arrange
        const text = "Rombit";
        const element = <A.Copyright name={text} />;

        // Act
        const copy = shallow(element);

        // Act and assert
        expect(copy.find(".a-copyright__label").text()).toBe(text);
    });
});
