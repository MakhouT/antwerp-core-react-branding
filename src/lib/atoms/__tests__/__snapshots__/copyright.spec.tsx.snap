// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copyright Renders default 1`] = `
<div
  className="a-copyright"
>
  <div
    className="a-copyright__sign"
  >
    <span
      className="fa fa-copyright"
    />
    2017
  </div>
  <span
    className="a-copyright__label"
  >
    Rombit
  </span>
</div>
`;
